### PR TITLE
dingtalk: init at 7.6.15.4102301

### PR DESCRIPTION
**Pull request by xddxdd**
DingTalk is an enterprise communication and collaboration platform developed by Alibaba Group.

Fixes #155902

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/di/dingtalk/package.nix [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    # Cleanup
+    rm -f release/{*.a,*.la,*.prl}
+    rm -f release/dingtalk_crash_report
+    rm -f release/dingtalk_updater
+    rm -f release/libapr*
+    rm -f release/libcrypto.so.*
+    rm -f release/libcurl.so.*
+    rm -f release/libdouble-conversion.so.*
+    rm -f release/libEGL*
+    rm -f release/libfontconfig*
```
Comment (FliegendeWurst) [inline pkgs/by-name/di/dingtalk/package.nix:179]:
```suggestion
    rm release/lib{apr,crypto,curl,double-conversion,EGL,fontconfig}*
```
And so on. You can remove at least 10 libraries per line.

File: pkgs/by-name/di/dingtalk/package.nix [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+      extraConfig = {
+        "Name[zh_CN]" = "钉钉";
+        "Name[zh_TW]" = "钉钉";
+      };
+    })
+  ];
+
+  meta = {
+    maintainers = with lib.maintainers; [ xddxdd ];
+    description = "DingTalk is an enterprise communication and collaboration platform developed by Alibaba Group";
```
Comment (FliegendeWurst) [inline pkgs/by-name/di/dingtalk/package.nix:265]:
```suggestion
    description = "Enterprise communication and collaboration platform developed by Alibaba Group";
```

File: pkgs/by-name/di/dingtalk/package.nix [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+  libthai,
+  libxcrypt-legacy,
+  libxkbcommon,
+  mesa,
+  mtdev,
+  nghttp2,
+  nspr,
+  nss,
+  openldap,
+  openssl_1_1,
```
Comment (FliegendeWurst) [inline pkgs/by-name/di/dingtalk/package.nix:52]: OpenSSL 1.1 will be dropped #269713

**Review by FliegendeWurst — CHANGES_REQUESTED** [review]

File: pkgs/by-name/di/dingtalk/package.nix (hunk 3 above)
Comment (xddxdd) [reply pkgs/by-name/di/dingtalk/package.nix:52]: Thanks for letting me know. Since OpenSSL 1.1 is a hard dependency for DingTalk, I will close this PR.

**Pull request closed without merging**